code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -325,7 +325,7 @@
 
 /obj/machinery/porta_turret/take_damage(damage, damage_type = BRUTE, damage_flag = 0, sound_effect = 1)
 	. = ..()

```

Comment:
i'm suprised that take damage does not manage qdeleted/integrity < 0 higher up the call chain?

Reply:
https://github.com/tgstation/tgstation/blob/34e523fd1f5d6b6593539b065a619507716dd666/code/game/objects/obj_defense.dm#L3-L22

Reply:
I guess the problem is that you can take damage at the point of break.

Perhaps set the returned damage to negative or zero if the object is broken?

Reply:
Not sure if that's gonna break every child with spaghetti code though. Can't be sure without going through them all and that's gonna be a drag to go through, but I'll do it if you want that solution.

Reply:
you know what, just move the parent call to the bottom of the proc, below the spark system generation and handling.

easy

Reply:
But then it will skip the if cause the return variable is null. And you will also get a callback to a qdeleted object, which given isn't much of a problem seeing all the thousands of callbacks that happen daily, but still.

Reply:
yeah

Reply:
Well, I'll leave it at this then. Something else needs to be done for that if to move the parent call at the end, but it would probably be a lot of work because the whole point is to see if you actually take damage in order to spark and you can't do that without refactoring the parent call which is what tells you whether there is damage or not.